org-code-javabuilder/README.md

```diff
@@ -1,33 +1,138 @@
+* [protocol](https://github.com/code-dot-org/javabuilder/tree/main/org-code-javabuilder/protocol)
+  contains the communication protocol that handles messages between Java Lab (client) and
+  Javabuilder (server)
+* [theater](https://github.com/code-dot-org/javabuilder/tree/main/org-code-javabuilder/theater)
+  contains a package that can be imported by the user to interact with the theater 
+  mini-app, an image and sound editing app created for AP CSA and used in standalone Java
+  Lab projects.
+
+## Local Development
+### Setup your project
+1. Install a Java IDE - We recommend IntelliJ Community edition.
+    1. If given the option when installing, select Java 11 as the default
+1. If Java 11 was not installed as part of the IDE, install 
+   [OpenJDK 11](https://jdk.java.net/java-se-ri/11)
+1. Download the repo and build. Building is a one-time step that seeds the git commit
+   hooks.
+```
+git clone git@github.com:code-dot-org/javabuilder.git
+cd javabuilder/org-code-javabuilder
+./gradlew build
+```
+That's it! Now you can open the project in your IDE and start developing. Here are some
+other useful commands:
+
+**Launch the local Javabuilder WebSocket server**
+```
+./gradlew appRun
+```
+*Note: the Javabuilder server will be ready to use when you see
+`Press any key to stop the server.` It will never reach `100% EXECUTING`.*
+
+**Run all tests**
+```
+./gradlew test
+```
+
+**Run the linter and fix lint errors**
+```
+./gradlew goJF
+```
+
+### Developing with Dashboard (most common scenario)
 In order to run Java Lab (Code Studio client) with Javabuilder, use the WebSocketServer.
 This is a local replacement of AWS API Gateway. 
 
-Instructions (Note: this method is still in development)
-1. Launch the WebSocketServer: `gradle appRun`
-1. Launch dashboard using the instructions here: https://github.com/code-dot-org/code-dot-org/blob/staging/SETUP.md#overview 
-1. Navigate to any Java Lab level, for example: http://localhost-studio.code.org:3000/projects/javalab/new
+Instructions
+1. Launch the WebSocketServer using the instructions above. (run `./gradlew appRun`)
+1. Launch dashboard using the instructions here: 
+   https://github.com/code-dot-org/code-dot-org/blob/staging/SETUP.md#overview 
+1. Navigate to any Java Lab level, for example: 
+   http://localhost-studio.code.org:3000/projects/javalab/new
 1. Click the "Run" button
 
 Alternatively, you can test in "headless" mode
 1. Install wscat: `npm install -g wscat`
-1. Launch the WebSocketServer: `gradle appRun`
-1. In a terminal, connect to the server with wscat: `wscat -c ws://localhost:8080/javabuilder`
+1. Launch the WebSocketServer: `./gradlew appRun`
+1. In a terminal, connect to the server with wscat: 
+   `wscat -c ws://localhost:8080/javabuilder`
+
+Postman is a good wscat alternative. It is in beta, but has a gui and is more
+feature-rich.
+
+### Developing with AWS
+For development purposes, you generally shouldn't need to deploy Javabuilder. Some cases
```

Comment:
should we instead link to the other readme for this so it isn't duplicated? I think you can link to a specific heading

Reply:
👍